File: apps/python-src/taskRunner.py
import time
import random

def sleep(minDelay, maxDelay=None):
    if maxDelay is not None:
        minDelay = random.random() * (maxDelay - minDelay) + minDelay
    time.sleep(minDelay)

def simulateOperation(minDelay, maxDelay=None, failure=None):
    if failure is not None and random.random() < failure:
        raise ValueError('Operation failed!')
    sleep(minDelay, maxDelay)

File: apps/nodejs-src/taskRunner.js
const sleep = async (msDelay, maxDelay = undefined) => {
    return new Promise((resolve, reject) => {
        if (maxDelay) {
            msDelay = Math.random() * (maxDelay - msDelay) + msDelay
        }
        console.log(msDelay)
        setTimeout(() => { resolve() }, msDelay)
	})
}

exports.simulateOperation = async (msDelay, 
    maxDelay = undefined, failure = undefined) => {
        if (failure && Math.random() < failure)
            throw new Error(`Operation failed!`)
        await sleep(msDelay, maxDelay)
}